cli/lsp/config.rs

```diff
@@ -15,19 +15,57 @@ pub struct ClientCapabilities {
   pub workspace_did_change_watched_files: bool,
 }
 
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CodeLensSettings {
+  #[serde(default)]
+  pub references: bool,
+  #[serde(default)]
+  pub references_all_functions: bool,
+}
+
 #[derive(Debug, Clone, Default, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct WorkspaceSettings {
   pub enable: bool,
   pub config: Option<String>,
   pub import_map: Option<String>,
+  pub code_lens: Option<CodeLensSettings>,
 
   #[serde(default)]
   pub lint: bool,
   #[serde(default)]
   pub unstable: bool,
 }
 
+impl WorkspaceSettings {
+  /// Determine if any code lenses are enabled at all.  This allows short
+  /// circuiting when there are no code lenses enabled.
+  pub fn enabled_code_lens(&self) -> bool {
+    if let Some(code_lens) = &self.code_lens {
+      code_lens.references
+    } else {
+      false
+    }
+  }
+
+  pub fn enabled_code_lens_references(&self) -> bool {
+    if let Some(code_lens) = &self.code_lens {
+      code_lens.references
+    } else {
+      false
+    }
+  }
```

Comment:
Isn't this the same as `enabled_code_lens`? Confused.

Reply:
It is at the moment, because we only have one source of code lenses.  But the next one is implementations, and that means `enabled_code_lens` will have to consider `implementations` and there will be a `enabled_code_lens_implementations` method.

Reply:
I will make that clear in the code doc.